preprocessing/science/kinematic_models/FaultPlane.py

```diff
@@ -374,22 +375,47 @@ def init_from_srf(self, fname):
                             self.aSR[j, i, 0:ndt1] = np.array([float(v) for v in lSTF])
                             break
 
-    def assess_STF_parameters(self):
+    def assess_STF_parameters(self, threshold):
         "compute rise_time (slip duration) and t_acc (peak SR) from SR time histories"
+        assert threshold >= 0.0 and threshold < 1
         self.rise_time = np.zeros((self.ny, self.nx))
         self.tacc = np.zeros((self.ny, self.nx))
+        misfits_Yoffe = []
+        misfits_Gaussian = []
         for j in range(self.ny):
             for i in range(self.nx):
                 if not self.slip1[j, i]:
                     self.rise_time[j, i] = np.nan
                     self.tacc[j, i] = np.nan
                 else:
-                    first_non_zero = np.amin(np.where(self.aSR[j, i, :])[0])
-                    last_non_zero = np.amax(np.where(self.aSR[j, i, :])[0])
-                    id_max = np.where(self.aSR[j, i, :] == np.amax(self.aSR[j, i, :]))[0]
+                    id_max = np.where(self.aSR[j, i, :] == np.amax(self.aSR[j, i, :]))[0][0]
+                    peakSR = self.aSR[j, i, id_max]
+                    ids_greater_than_threshold = np.where(self.aSR[j, i, :] > threshold * peakSR)[0]
+                    first_non_zero = np.amin(ids_greater_than_threshold)
+                    last_non_zero = np.amax(ids_greater_than_threshold)
                     self.rise_time[j, i] = (last_non_zero - first_non_zero + 1) * self.dt
                     self.tacc[j, i] = (id_max - first_non_zero + 1) * self.dt
                     self.t0[j, i] += first_non_zero * self.dt
+                    # 2 dims: 0: Yoffe 1: Gaussian
+                    newSR = np.zeros((self.ndt, 2))
+                    ts = self.tacc[j, i] / 1.27
```

Comment:
Where does the 1.27 come from?

Reply:
Yes, please add a new variable for this "magic number". Thanks!

Reply:
"It seems convenient to introduce a new parameter Tacc, which is defined as the time to peak slip velocity (i.e., the duration of positive slip acceleration) as illustrated in Figure 2. In the original Yoffe function, Tacc is zero because of the singularity. In contrast, in the regularized Yoffe function, Tacc is controlled by the duration of the triangular function used as the smoothing operator τS which is not a physical parameter. The temporal smoothing of the original Yoffe function yields finite peak-slip-velocity values. Because τS appears only in the regularized Yoffe function, we numerically investigated the relation between Tacc and τS for different τR. As shown in Figure 3, Tacc is linearly related to τS, confirming that τS directly controls the duration of the positive slip acceleration (Tacc). In particular, the ratio Tacc/τS does not depend on other parameters and is equal to 1.27 ± 0.01. The linear relation has been inferred by varying the rise time between 0.2 and 6.0 sec. Figure 3 clearly shows that τR does not affect Tacc."
From Tinti et al. 2005
I will add a comment.